benbarnett02.CodeTrack/CodeTrack/Program.cs

```diff
@@ -0,0 +1,310 @@
+﻿using Spectre.Console;
+using System.Globalization;
+namespace TrackingProgram;
+
+// This class handles all user interface elements and input validation.
+class Program
+{
+    // Configuration manager, as required. Just for connection string at the moment.
+    static void Main(string[] args)
+    {
+        TrackingData.StartupDatabase();
+        // Pretend like it takes a long time to connect to a local database, lol.
+        Console.WriteLine("-------------------------------\n");
+        Console.WriteLine("Connecting to " + TrackingData.connectionString);
+        Console.WriteLine("\n-----------------------------\n");
+        Console.WriteLine("Welcome to Code Tracker! \n");
+
+        MainMenu();
+    }
+
+    static void MainMenu()
+    {
+        bool apprun = true;
+        while (apprun)
+        {
+            Console.WriteLine("Please choose from the following options:");
+            Console.WriteLine("\t1. Add time entry manually");
+            Console.WriteLine("\t2. Start stopwatch");
+            Console.WriteLine("\t3. View/Edit past time entries");
+            Console.WriteLine("(type the number you want to select, then press enter)");
+
+
+            string mainMenuSelection = Console.ReadLine();
+
+            switch (mainMenuSelection)
+            {
+                case ("0"):
+                    apprun = false;
+                    break;
+
+                case ("1"):
+                    ManualEntry();
+                    break;
+
+                case ("2"):
+                    Stopwatch();
+                    break;
+
+                case ("3"):
+                    PastEntries();
+                    break;
+
+                default:
+                    break;
+            }
+        }
+    }
+
+    static void ManualEntry()
+    {
+        // Initialisation stuff for datetime parsing.
+        string format = "dd/MM/yyyy HH:mm tt";
+        CultureInfo culture = CultureInfo.CreateSpecificCulture("en-AU");
+
+        // Initialise these variables to have them in the scope of the method, not the loops below... made that mistake.
+        DateTime startDateTime;
+        DateTime endDateTime;
+
+        // Avoiding nulls, probably pointless but not sure.
+        string enteredStartTime = DateTime.Now.ToString(format);
```

Comment:
Looks like these aren't being used. Always write as little code as possible. Please remove of any unused statements/declarations. 😁